<?xml version="1.0"?>
<launch>
    <!-- Robot namespace to control -->
    <arg name="robot" default="L1" />

    <!-- Controller Parameters Arguments -->
    <arg name="control_frequency" default="30.0" />

    <arg name="pose_ns" default="vrpn_client_node" />

    <!-- Topic names -->
    <arg name="cmd_vel_topic" default="$(arg robot)/cmd_vel" />
    <arg name="goal_topic" default="$(arg robot)/goal" />
    <arg name="pose_topic" default="$(arg pose_ns)/$(arg robot)/pose" />


    <!-- Run diff_control Node with gdb -->
    <node pkg="diff_control" type="diff_control_node" name="diff_controller" output="screen">
        <param name="pose_topic" value="$(arg pose_topic)" />
        <param name="goal_topic" value="$(arg goal_topic)" />

        <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)" />

        <!-- Controller Parameters -->
        <param name="a" value="0.15" />
        <param name="Kx" value="1" />
        <param name="Ky" value="1" />
        <param name="control_frequency" value="$(arg control_frequency)" />
    </node>


</launch>